dig.go

```diff
@@ -372,10 +390,17 @@ func (c *Container) provide(ctor interface{}, opts provideOptions) error {
 			return err
 		}
 	}
+	ge := n.toGraphEdges()
+	c.addToGraph(ge)
```

Comment:
Since the `ge` variable is only used in the call here, we could instead have a `Container.addDotNode` method that accepts the `node` struct.

```
func (c *Container) addDotNode(n *node) {
  c.dotgraph = append(c.dotgraph, n.toGraphEdges()...)
}
```